### PR TITLE
[1364] Add support for a basic Image view widget

### LINKED ISSUE #1364: Add support for a basic Image view widget
A basic Image widget which simply displays an image from an arbitrary URL (computed by the backend).
- No user interaction on it, just displaying an ``.
- Can be used to display images hosted by the backend itself (i.e. `/api/images/*`), but could possibly be used to display images from anywhere else.
- Available in the core, in the View DSL and in the visual form editor.

### PR DESCRIPTION
**Pull request by pcdavid**
# Pull request template

An example usage can be tested when configuring an "Image" style for a Node Description:

https://user-images.githubusercontent.com/10608/193613173-6f02645d-ee7f-4313-ae6b-9ccd6c068bfb.mp4

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [x] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [x] Does the title of the commit contributing the ADR start with `[doc]`?
- [x] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [x] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [x] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Image.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            this.label = Objects.requireNonNull(label);
+            return this;
+        }
+
+        public Builder iconURL(String iconURL) {
+            this.iconURL = Objects.requireNonNull(iconURL);
+            return this;
+        }
+
+        public Builder url(String value) {
```
Comment (AxelRICHARD) [inline packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Image.java]: value -> url

Comment (AxelRICHARD) [inline packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Image.java]: value -> url

File: packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/ImageDescription.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        }
+
+        public Builder messageProvider(Function<Object, String> messageProvider) {
+            this.messageProvider = Objects.requireNonNull(messageProvider);
+            return this;
+        }
+
+        public ImageDescription build() {
+            ImageDescription imageDescription = new ImageDescription();
+            imageDescription.id = this.id;
```
Comment (AxelRICHARD) [inline packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/ImageDescription.java]: Please add Objects.requireNonNull(this.id)

**Review by AxelRICHARD — CHANGES_REQUESTED** [review]

File: packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Image.java (hunk 1 above)
Comment (pcdavid) [reply packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Image.java]: done

File: packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/ImageDescription.java (hunk 2 above)
Comment (pcdavid) [reply packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/ImageDescription.java]: done

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**